flink-contrib/flink-storm-compatibility/flink-storm-compatibility-core/src/main/java/org/apache/flink/stormcompatibility/api/FlinkTopologyBuilder.java

```diff
@@ -138,6 +138,21 @@ public FlinkTopology createTopology() {
 					DataStream<?> inputDataStream = availableOperators.get(producerId);
 
 					if (inputDataStream != null) {
+						// get upstream operator
```

Comment:
Move this code with the `if (grouping.is_set_fields())` then-Block (line 161). Introduce local variable and look up declarer in introduced HashMap: `FlinkOutputFieldsDeclarer d = this.declarers.get(producerId)`